### PR TITLE
Add remove punctuation, remove stopwords and case-sensitive buttons

### PR DESCRIPTION
**Pull request by alexcombessie**
[CH62119](https://app.clubhouse.io/dataiku/story/62119/add-remove-punctuation-remove-stopwords-and-case-sensitive-buttons)

### CODE REVIEW

File: python-lib/wordcloud_visualizer.py [hunk 1]
```diff
@@ -208,6 +211,27 @@ def _tokenize_texts(self, df_grouped: List) -> List:
...
+        It then selects the case version with the highest count to represent the whole group.
+        In case of a tie, the lowercase version is chosen.
+
+        Args:
+            counts: token counter e.g., Counter({"The": 3, "the": 5, "best": 3, "Best": 4})
+        Returns:
+            New token counter with a single case version for each token and the sum of counts across case versions
+                e.g., Counter({"the": 8, "Best": 7})
+        """
+        df_counts = pd.DataFrame.from_dict(counts.items())
```
Comment (Alexlandeau) [inline python-lib/wordcloud_visualizer.py:227]: Cf. comment on pandas 1 in code env desc.json and version support

File: resource/stopwords/ja.txt [hunk 2]
```diff
@@ -0,0 +1,111 @@
+あ
```
Comment (Alexlandeau) [inline resource/stopwords/ja.txt:1]: Japanese isn't supported, cf. https://github.com/dataiku/dss-plugin-nlp-visualization/blob/9572eb5e0540614f52cb739ae114022c7ffcbd6e/python-lib/language_dict.py#L67

File: resource/stopwords/ko.txt [hunk 3]
```diff
@@ -0,0 +1,55 @@
+가
```
Comment (Alexlandeau) [inline resource/stopwords/ko.txt:1]: Korean isn't supported, cf. https://github.com/dataiku/dss-plugin-nlp-visualization/blob/9572eb5e0540614f52cb739ae114022c7ffcbd6e/python-lib/language_dict.py#L67

**Review by Alexlandeau — CHANGES_REQUESTED** [review]

File: code-env/python/desc.json [hunk 4]
```diff
@@ -1,8 +1,10 @@
...
     "acceptedPythonInterpreters": [
-        "PYTHON36"
+        "PYTHON36",
+        "PYTHON37"
     ],
     "forceConda": false,
     "installCorePackages": true,
-    "installJupyterSupport": true
+    "installJupyterSupport": true,
+    "corePackagesSet": "PANDAS10"
```
[the remark this answers is not in the document]
Comment (alexcombessie) [reply code-env/python/desc.json:9]: DSS 7 will no longer be supported in a month, so I think we can drop it already

[the remark this answers is not in the document]
Comment (alexcombessie) [reply code-env/python/desc.json:9]: It creates a incentive for customers to upgrade, which is probably a good thing

File: custom-recipes/nlp-visualization-wordcloud/recipe.json [hunk 5]
```diff
@@ -65,6 +65,35 @@
...
         },
+        {
+            "name": "separator_text_handling",
+            "label": "Text simplification",
+            "type": "SEPARATOR"
+        },
+        {
+            "name": "remove_stopwords",
+            "label": "Remove stopwords",
+            "description": "Remove common words with little meaning e.g., the, a, be, have",
```
[the remark this answers is not in the document]
Comment (alexcombessie) [reply custom-recipes/nlp-visualization-wordcloud/recipe.json:76]: Good idea

File: python-lib/wordcloud_visualizer.py (hunk 1 above)
Comment (alexcombessie) [reply python-lib/wordcloud_visualizer.py:227]: Yes, this is pandas 1.0 only. I guess that's fine as we won't support DSS 7 much longer

File: resource/stopwords/ja.txt (hunk 2 above)
Comment (alexcombessie) [reply resource/stopwords/ja.txt:1]: Agreed, but this issue will soon be solved (cf https://github.com/WorksApplications/SudachiPy/issues/148)
So I included this file in preparation of a next version which will add Japanase

File: resource/stopwords/ko.txt (hunk 3 above)
Comment (alexcombessie) [reply resource/stopwords/ko.txt:1]: Yep, good catch. It's less likely that we'll be able to solve the installation issues inherent to the Korean tokenization library (external open-source lib). I would rather keep the stopwords here, as it means we could provide Korean support to customers who request it, with a simple fork and some consulting help to install the Korean dependencies on the server

**Review by Alexlandeau — APPROVED** [review]

**Pull request merged**